### ISSUE TITLE
Images are replaced with placeholders too late in v0.2.7

### ISSUE THREAD

**Issue by kantaiz3** [silveredgold/beta-protection#65]
Hello! I just tested out v0.2.7 and found a major bug 🐛! You likely know about this already but I wanted to report this just in case.

All images are unblurred before they are replaced with placeholders in v0.2.7! The time between unblur and being replaced with a placeholder varies from page to page, but it's in the range of half a second (on simpler sites) to multiple seconds (e.g. reddit feed). This happens on all sites that I tested (reddit, booru, twitter, etc.).

I hope that this can be fixed soon. Thank you!!

**Comment by silveredgold** [comment]
v0.2.8 is out now and is the final attempt at this concept. Hopefully 0.2.8 is good enough, but be aware that if it causes any more bugs, it might be removed in a future release as it really does complicate the extension's logic and page lifecycle.

**Comment by kantaiz3** [comment]
I just tested 0.2.8 and it works amazingly! It nicely blurs out images on all websites that I tested and leaves allowed domains alone. (No delay or side effects like before!) 
The only thing I did notices was that tabs with direct images (no website, just the browser showing an image) did show the image for a fraction of a second (<0.1sec) before blurring and censoring it. But this is just a super minor thing. Otherwise, 0.2.8 is perfect! Thank you so much!!


Also, I really like the blur effect. Are there plan to have an option to not use placeholders at all and just keep the blur until the image is censored? I think it would make for a really nice effect.
